src/Jellyfin.Plugin.Dlna/ContentDirectory/ControlHandler.cs

```diff
@@ -1023,18 +1011,45 @@ private QueryResult<ServerItem> GetLatest(BaseItem parent, InternalItemsQuery qu
     {
         query.OrderBy = Array.Empty<(ItemSortBy, SortOrder)>();
 
+        int limit;
+
+        if (query.StartIndex > 0)
+        {
+            limit = (query.Limit is null || query.Limit <= 0) ? int.MaxValue : (query.StartIndex + query.Limit).Value;
+        }
+        else
+        {
+            limit = query.Limit ?? 50;
+        }
+
         var items = _userViewManager.GetLatestItems(
             new LatestItemsQuery
             {
                 // User cannot be null here as the caller has set it
                 User = query.User!,
-                Limit = query.Limit ?? 50,
+                Limit = limit,
                 IncludeItemTypes = new[] { itemType },
                 ParentId = parent?.Id ?? Guid.Empty,
                 GroupItems = true
             },
             query.DtoOptions).Select(i => i.Item1 ?? i.Item2.FirstOrDefault()).Where(i => i is not null).ToArray();
 
+        if (query.StartIndex > 0)
+        {
+            if (items.Length <= query.StartIndex)
+            {
+                items = Array.Empty<BaseItem>();
+            }
+            else if (query.Limit > 0 && items.Length > query.Limit.Value)
+            {
+                items = items[query.StartIndex.Value..(query.StartIndex + query.Limit).Value];
```

Comment:
This can throw an out of bounds exception (if start index + limit > length)

Reply:
I believe I fixed it. I used the wrong limit variable... Again. This is supposed to be a sanity check to prevent the plugin from returning more than `query.Limit` elements if the server unexpectedly returns more items than requested. Now that I'm writing this, sounds kinda stupid. Should I remove it?